### PR TITLE
Integration tests for execution fee

### PR DESCRIPTION
**Pull request by ALFecki**
- Added tests for execution fee estimation
- Added baedeker running 3 instances of AssetHub for integration testing
- Added all setup files for executing CI tests

### CODE REVIEW

File: .baedeker/init_fee.ts [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+              {
+                Transact: {
+                  originKind: 'Superuser',
+                  requireWeightAtMost: {
+                    refTime: 8000000000,
+                    proofSize: 8000,
+                  },
+                  call: {
+                    encoded:
+                      '0x3201011f00d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d0104',
```
Comment (mrshiposha) [inline .baedeker/init_fee.ts:121]: make an extrinsic in code please

```suggestion
                      apiAssetHubA.tx.assets.forceCreate(1984, alice.address, true, 10000).method.toHex(),
```

Comment (mrshiposha) [reply .baedeker/init_fee.ts:121]: the same for the similar places

File: .baedeker/init_fee.ts [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+
+  console.log('Asset metadata on AssetHubC');
+
+  await new Promise(f => setTimeout(f, INTERVAL_ASHB));
+
+  await retry(() =>
+    apiAssetHubA.tx.assets
+      .mint(
+        1984,
+        {id: '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY'},
```
Comment (mrshiposha) [inline .baedeker/init_fee.ts:283]:
```suggestion
        {id: alice.address},
```

Comment (mrshiposha) [reply .baedeker/init_fee.ts:283]: The same for the similar places

File: .baedeker/init_fee.ts [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+          },
+        },
+        {
+          V4: {
+            parents: 0,
+            interior: {
+              X1: [
+                {
+                  AccountId32: {
+                    id: `0x${Buffer.from(decodeAddress(alice.address)).toString('hex')}`,
```
Comment (mrshiposha) [inline .baedeker/init_fee.ts:327]:
```suggestion
                    id: alice.addressRaw,
```

Comment (mrshiposha) [reply .baedeker/init_fee.ts:327]: The same for the similar places

File: packages/xcm-estimate/estimator.ts [hunk 4]
```diff
@@ -293,6 +293,7 @@ export class Estimator {
     );
 
     if (errors.length > 0) {
+      console.log(errors);
```
Comment (mrshiposha) [inline packages/xcm-estimate/estimator.ts:296]: Why is it here?

File: tests/integration/fee-estimation.test.ts [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        },
+      ]),
+    );
+
+  await registry.addNativeCurrency('Polkadot');
+  const xcmAssetHubA = await registry.connectXcm('AssetHubA');
+  const xcmAssetHubB = await registry.connectXcm('AssetHubB');
+  const xcmAssetHubC = await registry.connectXcm('AssetHubC');
+
+  test('correct composeTransfer: A -> B', async () => {
```
Comment (mrshiposha) [inline tests/integration/fee-estimation.test.ts:66]: Please add tests that use the estimator directly. Namely, the `tryEstimateExtrinsicFees`. And use `chainXcmFeesLogger` to check that we estimated fees on remote chains

**Review by mrshiposha — CHANGES_REQUESTED** [review]

File: .baedeker/init_fee.ts (hunk 2 above)
Comment (ALFecki) [reply .baedeker/init_fee.ts:283]: yes, I did that, but the script doesn't work correctly after that, so I left it until the tests were fixed.

File: packages/xcm-estimate/estimator.ts (hunk 4 above)
Comment (ALFecki) [reply packages/xcm-estimate/estimator.ts:296]: For debugging, PR is in draft yet :)

**Pull request merged**